tools/shared.py

```diff
@@ -1332,6 +1338,145 @@ def __getitem__(self, key):
     return self.instance()[key]
 
 
+class settings_parser:
```

Comment:
Classes are normally spelt like "SettingsParser".

Also, rather than putting more stuff in shared.py what do you think about putting this in a new file?